keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,920 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The authors have taken inspiration from the [CSI development](https://github.com/kubernetes/design-proposals-archive/blob/main/storage/container-storage-interface.md). 
+Kubernetes compute management is tightly integrated with the Kubelet and the 
+existing suite of resources managers including the topology manager, CPU manager, 
+memory manager, and device manager. While these managers have added functionality that 
+has addressed a varied set of use cases, they do present the community with several challenges.
+
+Adding a new capability to one of these managers is slow moving and difficult 
+due to their complex interactions, the level of prudence required given the potential
+impacts, the implied necessity to make the extension work in as many scenarios as
+possible, as well as the overhead heaped onto the sig-node maintainers. More 
+details on the challenges discussed with the community have been captured in 
+the [CPU Management Kubelet Use Cases & Current State document](https://docs.google.com/document/d/1U4jjRR7kw18Rllh-xpAaNTBcPsK5jl48ZAVo7KRqkJk).  
+
+As a result, adding optimizations necessary to support some more niche use cases, 
+to improve total cost of ownership for deployments with more exacting requirements 
+on resources or to provide a mechanism to progressively roll-out the benefits of 
+innovations related to newer CPU architectures is difficult and high cost to community resources.
+
+This proposal aims to address the challenges by introducing a new Container Compute 
+Interface (CCI). The CCI is conceptually equivalent to the CSI in that it will 
+delineate between the responsibility of the Kubelet and lower-level responsibility
+of compute focused plugins for resource assignment capabilities, such CPU and memory. 
+
+We are propsing an iterative implementation approach: in the first stage (alpha) we 
+propose to guarantee the exclusive operation of CCI Manager on cpu and memory resources
```

Comment:
In general, this paragraph needs some work to be more crisp in its explanation of what is being proposed.

For instance, this is the first time the term CCIManager is introduced, and the explanation around why one needs to set the `none` policy on the CPUManager is unclear.

I would probably word this as:

> We propse an iterative implementation approach:
>
> In the first stage (alpha) we propose to add a new component to the `kubelet` called the `CCIManager` that focuses exclusively on providing a mechanism for pluggable CPU management policies (as a replacement to the builtin `CPUManager` and its corresponding policies). Enablement of the `CCIManager` will be blocked by a feature gate, which, when enabled, will trigger all CPU Management decisions to flow through the `CCIManager` instead of the builtin `CPUManager`. As such, the `CPUManager` policy MUST be set to `none` if this new feature gate is enabled. This will be enforced by the kubelet, causing it to "fail fast" if this invariant does not hold true.
>
> For now, only a _single_ plugin will be allowed to run on a node at any given time. Details of how we plan to enforce this requirement as well as how we plan to (temporarily) handle bootstrapping a node that does not yet have a pluggable CPU Management policy running can be found in Section [X]. In a later phase, we will provide a proper mechanism to handle bootstrapping as well as cases where a plugin becomes temporarily unavailable.
```

Reply:
done